src/Response.php

```diff
@@ -144,7 +144,7 @@ public function getReasonPhrase(): string
     /**
      * {@inheritdoc}
      */
-    public function withStatus($code, $reasonPhrase = ''): Response
+    public function withStatus($code, $reasonPhrase = ''): static
     {
```

Comment:
If the docblock already states `@return static`, then this patch shouldn't be necessary for static analysis purposes

Reply:
It may be an issue in phpstan then.
Let me provide a wider context with a code example, and then I'll shut up :smiley:  and search for a solution from that POV.

```
use Psr\Http\Message\ResponseInterface;
use Laminas\Diactoros\Response\JsonResponse;

class DecoratedResponse implements ResponseInterface {
    private JsonResponse $jsonResponse;

    /* A lot of other method delegation to $this->jsonResponse */

    public function withStatus($code, $reasonPhrase = ''): static
    {
        $newInstance = clone $this;
        $newInstance->jsonResponse = $this->jsonResponse->withStatus($code, $reasonPhrase);
        return $newInstance;
    }
}
```
This delegation doesn't return the expected type:

`DecoratedResponse::$jsonResponse (Laminas\Diactoros\Response\JsonResponse) does not accept Laminas\Diactoros\Response.`

Reply:
Can you perhaps try changing `{@inheritDoc}` to `@inheritDoc`? Perhaps phpstan is not prioritizing the docblock.

Reply:
It didn't do much. I changed it to `@return static` then it worked. So I guess phpstan doesn't handle inheriting type information from docblock well.